solc/CommandLineInterface.cpp

```diff
@@ -1275,41 +1275,54 @@ void CommandLineInterface::outputCompilationResults()
 {
 	solAssert(CompilerInputModes.count(m_options.input.mode) == 1);
 
-	handleCombinedJSON();
-
-	// do we need AST output?
-	handleAst();
-
-	CompilerOutputs astOutputSelection;
-	astOutputSelection.astCompactJson = true;
-	if (m_options.compiler.outputs != CompilerOutputs() && m_options.compiler.outputs != astOutputSelection)
+	try
 	{
-		// Currently AST is the only output allowed with --stop-after parsing. For all of the others
-		// we can safely assume that full compilation was performed and successful.
-		solAssert(m_options.output.stopAfter >= CompilerStack::State::CompilationSuccessful);
+		handleCombinedJSON();
 
-		for (std::string const& contract: m_compiler->contractNames())
+		// do we need AST output?
+		handleAst();
+
+		CompilerOutputs astOutputSelection;
+		astOutputSelection.astCompactJson = true;
+		if (m_options.compiler.outputs != CompilerOutputs() && m_options.compiler.outputs != astOutputSelection)
 		{
-			if (needsHumanTargetedStdout(m_options))
-				sout() << std::endl << "======= " << contract << " =======" << std::endl;
-
-			handleEVMAssembly(contract);
-
-			if (m_options.compiler.estimateGas)
-				handleGasEstimation(contract);
-
-			handleBytecode(contract);
-			handleIR(contract);
-			handleIRAst(contract);
-			handleIROptimized(contract);
-			handleIROptimizedAst(contract);
-			handleSignatureHashes(contract);
-			handleMetadata(contract);
-			handleABI(contract);
-			handleStorageLayout(contract);
-			handleNatspec(true, contract);
-			handleNatspec(false, contract);
-		} // end of contracts iteration
+			// Currently AST is the only output allowed with --stop-after parsing. For all of the others
+			// we can safely assume that full compilation was performed and successful.
+			solAssert(m_options.output.stopAfter >= CompilerStack::State::CompilationSuccessful);
+
+			for (std::string const& contract: m_compiler->contractNames())
+			{
+				if (needsHumanTargetedStdout(m_options))
+					sout() << std::endl << "======= " << contract << " =======" << std::endl;
+
+				handleEVMAssembly(contract);
+
+				if (m_options.compiler.estimateGas)
+					handleGasEstimation(contract);
+
+				handleBytecode(contract);
+				handleIR(contract);
+				handleIRAst(contract);
+				handleIROptimized(contract);
+				handleIROptimizedAst(contract);
+				handleSignatureHashes(contract);
+				handleMetadata(contract);
+				handleABI(contract);
+				handleStorageLayout(contract);
+				handleNatspec(true, contract);
+				handleNatspec(false, contract);
+			} // end of contracts iteration
+		}
+	}
+	catch (UnimplementedFeatureError const& _unimplementedError)
+	{
+		SourceReferenceFormatter formatter(serr(false), *m_compiler, coloredOutput(m_options), m_options.formatting.withErrorIds);
+		m_hasOutput = true;
+		formatter.printExceptionInformation(
+			_unimplementedError,
+			Error::errorSeverity(Error::Type::UnimplementedFeatureError)
+		);
+		solThrow(CommandLineExecutionError, "");
 	}
```

Comment:
Note that we also have a top-level handler for this error in `main.cpp`: https://github.com/ethereum/solidity/blob/8a97fa7a1db1ec509221ead6fea6802c684ee887/solc/main.cpp#L48-L53

It will now be redundant and I'd remove it. `CommandLineInterface` is the top-level class and anything we consider a feature should be somewhere inside it. `UnimplementedFeatureError` outside of it is not really expected and we can let it end up in the "uncaught exception" handler. If that ever happens, it will likely be the result of a bug.

Reply:
>Though you should probably add handlers in `YulStack`, similar to the ones in `CompilerStack` 

I added `try/catch` blocks to `YulStack`, but I am not too sure if that's really what you were suggesting here.

Reply:
Yeah, it's fine the way you did it.